controllers/keystoneapi_controller.go

```diff
@@ -819,22 +814,23 @@ func (r *KeystoneAPIReconciler) ensureFernetKeys(
 // createHashOfInputHashes - creates a hash of hashes which gets added to the resources which requires a restart
 // if any of the input resources change, like configs, passwords, ...
 //
+// returns the hash, whether the hash changed (as a bool) and any error
+//
 func (r *KeystoneAPIReconciler) createHashOfInputHashes(
 	ctx context.Context,
 	instance *keystonev1.KeystoneAPI,
 	envVars map[string]env.Setter,
-) (string, error) {
+) (string, bool, error) {
+	var hashMap map[string]string
+	changed := false
 	mergedMapVars := env.MergeEnvs([]corev1.EnvVar{}, envVars)
 	hash, err := util.ObjectHash(mergedMapVars)
 	if err != nil {
-		return hash, err
+		return hash, changed, err
```

Comment:
shouldn't it retutn nil instead of changed=false? an error means no changed, not unchanged state

Reply:
We could change the bool to `*bool` and then be able to return `nil`, but as it at the moment, we have to return either true or false.